app/Collection/Collection.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Notes\Collection;
+
+use Notes\Model\NoteTag as NoteTagModel;
+
+class Collection implements \Iterator
+{
+    private $total = 0;
+    private $count=0;
+    private $pointer = 0;
+    private $objects = array();
+    
+    public function __construct($resultset = null)
+    {
+        $this->count = count($resultset);
+    }
+    
+    public function add($object)
+    {
+        if (is_object($object)) {
+            $this->objects[$this->total++] = $object;
+        } else {
+            throw new \InvalidArgumentException("Input should be Object");
+        }
+        
+    }
+
+    public function getResult($index)
+    {
+        if ($index >= $this->total || $index < 0) {
```

Comment:
should throw exception item not found exception. If index is greater than size of collection then should out of bound exception.